docs/proposals/federated-ruler.md

```diff
@@ -0,0 +1,70 @@
+
+## Challenges
+
+### Allow federated rules behind feature flag
+
+#### Challenge
+
+Federated tenant rules and alerts will not be a good fit for organization and should be behind a feature flag.
+
+#### Proposal
+
+For federated rules owned by a single tenant, creation of federated rules (those sourcing data from multiple tenants) should be blocked behind the feature flag `tenant-federation.enabled`.
+
+To support composite tenants, if tenant federation is enabled for ruler and alertmanager via `tenant-federation.enabled`, then ruler use a `mergeQueryable` to aggregate the results of querying multiple tenants. The ruler and alertmanager APIs should be updated to always call `tenant.GetTenantIDs` instead of `tenant.GetTenantID`, which will use the MultiTenantResolver when tenant federation is enabled.
+
+### Where to store resulting series of federated rule
+
+#### Challenge
+
+A single tenant rule always stores produced series in the tenant where the rule exists. This 1 -> 1 mapping becomes a many -> 1 mapping for federated rules.
+
+#### Proposal
+
+As some use cases will demand that a specific federated rule, querying tenants `B` and `C`, is stored in the owning team's tenant `A`, an option to allow explicit assignment of destination tenant for a federated rule is needed. Some use cases where a set of tenants `A|B|C|...|Z` are being treated as a single logical tenant when querying this explicit assignment of destination tenant is not needed.
+
+To allow both of these, we suggest an additional field to the rules proto `dest_tenant_id` which determines which tenant to send the series produced by the rule to, used if and only if the rule is a federated rule. If this field is not given, for composite tenants a random but consistent subtenant of the multiple tenants owning the rule is chosen using a hashmod of the series label to determine the subtenant and for single tenants owning a federated rule the resulting series is saved in the tenant which owns the rule.
+
+### Which tenants to query from for federated rules
+
+#### Challenge
+
+A single tenant rule always queries the tenant which owns the rule. This 1 -> 1 mapping becomes a 1 -> many mapping for federated rules.
+
+#### Proposal
+
+As some use cases will demand that a specific federated rule, querying tenant B and C, is stored in the owning teams tenant A, an option to allow explicit assignment of source tenants for a federated rule is needed. In the case of a composite tenant where a set of tenants `A|B|C|...|Z` are being treated as a single logical tenant when querying this explicit assignment of destination tenant is explicitly called for, but could prove useful.
```

Comment:
> when querying this explicit assignment of destination tenant is explicitly called for

what does this mean?

Reply:
it should read 

> when querying this explicit assignment of destination tenant isn't explicitly called for
since the set of tenants `A|..|Z` is being treated as one composite tenant by the system